### PR TITLE
Upgrade linter rules

### PR DESCRIPTION
**Pull request by arnaucube**

### CODE REVIEW

File: db/pebble/pebble.go [hunk 1]
```diff
@@ -102,15 +106,17 @@ func prefixIterOptions(prefix []byte) *pebble.IterOptions {
...
+func (p *Storage) Iterate(f func([]byte, []byte) (bool, error)) (err error) {
 	// NewIter already provides a point-in-time view of the current DB
 	// state, but if is used for long term (is not the case), should use an
 	// iterator over an snapshot:
 	// snapshot := p.pdb.NewSnapshot()
 	// defer snapshot.Close()
 	// iter := snapshot.NewIter(nil)
 	iter := p.pdb.NewIter(prefixIterOptions(p.prefix))
-	defer iter.Close()
+	defer func() {
```
Comment (ed255) [inline db/pebble/pebble.go:85]: This is problematic.  If there's an error in this function, we won't get the error but the result of `iter.Close()` which may be nil!  This is replacing the real error.  It should be something like:
```
err1 := iter.Close()
if err != nil {
  return err
} else {
  return err1
}
```

File: merkletree.go [hunk 2]
```diff
@@ -1013,9 +1049,11 @@ func RootFromProof(proof *Proof, k, v *big.Int) (*Hash, error) {
 			midKey = &HashZero
 		} else {
 			if bytes.Equal(kHash[:], proof.NodeAux.Key[:]) {
-				return nil, fmt.Errorf("Non-existence proof being checked against hIndex equal to nodeAux")
+				return nil,
+					fmt.Errorf("Non-existence proof being checked against hIndex equal to nodeAux")
 			}
-			midKey, err = LeafKey(proof.NodeAux.Key, proof.NodeAux.Value)
+			midKey, err = LeafKey(proof.NodeAux.Key,
```
Comment (ed255) [inline merkletree.go]: check if you really need to split this into two lines

File: merkletree.go [hunk 3]
```diff
@@ -802,14 +832,16 @@ func NewProofFromBytes(bs []byte) (*Proof, error) {
...
+		copy(p.NodeAux.Value[:],
+			nodeAuxBytes[ElemBytesLen:2*ElemBytesLen])
 	}
 	return p, nil
 }
 
 // Bytes serializes a Proof into a byte array.
 func (p *Proof) Bytes() []byte {
-	bsLen := proofFlagsLen + len(p.notempties) + ElemBytesLen*len(p.Siblings)
+	bsLen := proofFlagsLen + len(p.notempties) +
```
Comment (ed255) [inline merkletree.go]: check if you really need to split this into two lines

File: merkletree.go [hunk 4]
```diff
@@ -802,14 +832,16 @@ func NewProofFromBytes(bs []byte) (*Proof, error) {
 			return nil, ErrInvalidProofBytes
 		}
 		copy(p.NodeAux.Key[:], nodeAuxBytes[:ElemBytesLen])
-		copy(p.NodeAux.Value[:], nodeAuxBytes[ElemBytesLen:2*ElemBytesLen])
+		copy(p.NodeAux.Value[:],
```
Comment (ed255) [inline merkletree.go]: check if you really need to split this into two lines

**Review by ed255 — CHANGES_REQUESTED** [review]

File: db/pebble/pebble.go (hunk 1 above)
Comment (arnaucube) [reply db/pebble/pebble.go:85]: Right! Thanks!

**Review by ed255 — APPROVED** [review]

**Pull request merged**